File: sdk/src/main/java/co/omise/android/SDKLog.java
package co.omise.android;

/**
 * SDKLog is the custom SDK logging class that allows easier logging that is
 * specific to the SDK tag.
 */
public final class SDKLog {
    public static String TAG = "co.omise.android";

    public static void d(String message) {
        android.util.Log.d(TAG, message);
    }

    public static void e(String message, Throwable e) {
        android.util.Log.e(TAG, message, e);
    }

    public static void wtf(String message, Throwable e) {
        android.util.Log.wtf(TAG, message, e);
    }
}

File: sdk/src/main/java/co/omise/android/SDKLog.kt
package co.omise.android

import android.util.Log

/**
 * SDKLog is the custom SDK logging class that allows easier logging that is
 * specific to the SDK tag.
 */
internal object SDKLog {
    private const val TAG = "co.omise.android"

    fun d(message: String) {
        if (BuildConfig.DEBUG) {
            Log.d(TAG, message)
        }
    }

    fun e(
        message: String,
        e: Throwable,
    ) {
        if (BuildConfig.DEBUG) {
            Log.e(TAG, message, e)
        }
    }

    fun wtf(
        message: String,
        e: Throwable,
    ) {
        if (BuildConfig.DEBUG) {
            Log.wtf(TAG, message, e)
        }
    }
}
